src/app.ts

```diff
@@ -0,0 +1,57 @@
+type ProductTuple = {
+    name: string;
+    category: 'Elektronika' | 'Odzież' | 'AGD'; 
+    price: number;
+    isAvailable: boolean;
+    sales: number;
+};
+
+let products: ProductTuple[] = [
+    { name: "Smartfon", category: "Elektronika", price: 3000, isAvailable: true, sales: 11 },
+    { name: "Laptop", category: "Elektronika", price: 6000, isAvailable: false, sales: 15 },
+    { name: "T-shirt", category: "Odzież", price: 50, isAvailable: true, sales: 13 },
+    { name: "Bluza", category: "Odzież", price: 150, isAvailable: false, sales: 99 },
+    { name: "Lodówka", category: "AGD", price: 2000, isAvailable: true, sales: 22 },
+];
+
+function getAvailableProducts(array: ProductTuple[]): string[] {
+  let availableProducts = [];
+    availableProducts = array.filter((product) => product).map((product) => product.name);
+    return availableProducts;
+}
+
+function getProductsByCategory(array: ProductTuple[], { category }: { category: string }): ProductTuple[] {
+    const filteredArray = array.filter((element) => element.category === category);
+    return filteredArray;
+}
+
+function getAveragePriceByCategory(array: ProductTuple[], category: string): number {
+    const categoryProducts = getProductsByCategory(array, { category: category });
+    return categoryProducts.length
+        ? categoryProducts.reduce((sum, { price }) => sum + price, 0) / categoryProducts.length
+        : 0;
+}
+
+function sortProducts(
+    array: ProductTuple[],
+    key: "name" | "price" | "isAvailable",
+    bool: boolean = true
```

Comment:
Raczej inaczej bym nazwał ten parametr - na pierwszy rzut oka nie do końco rozumiałem o co chodzi ;)
Często używa się skrótów ASC oraz DESC - można użyć w formie enuma :)

https://www.bettercoder.io/job-interview-questions/403/what-is-the-meaning-of-asc-and-desc-keywords-in-sql-server